core/cmdbchangeop.class.inc.php

```diff
@@ -63,22 +63,53 @@ public static function Init()
 
 	/**
 	 * Describe (as a text string) the modifications corresponding to this change
-	 */	 
+	 */
 	public function GetDescription()
 	{
 		return '';
 	}
 
 	/**
-	 * Safety net: in case the change is not given, let's guarantee that it will
-	 * be set to the current ongoing change (or create a new one)	
-	 */	
+	 * @param string $sAttCode
+	 * @param mixed $value
+	 *
+	 * @return bool
+	 * @throws \CoreException
+	 * @throws \CoreUnexpectedValue
+	 *
+	 * @since 2.7.4 reject non persisted CMDBChange
+	 */
+	public function Set($sAttCode, $value)
+	{
+		if ($sAttCode !== 'change') {
+			return parent::Set($sAttCode, $value);
+		}
+
+		if ($value->IsNew()) {
+			throw new CoreUnexpectedValue("Cannot set CMDBChangeOp.change with a non persisted CMDBChange !");
+		}
```

Comment:
I would revert this logic so that the exception is thrown first and otherwise continue to to existing `return parent::Set()`
```suggestion
		if ($sAttCode == 'change' && $value->IsNew()) {
			throw new CoreUnexpectedValue("Cannot set CMDBChangeOp.change with a non persisted CMDBChange !");
		}
```

That would improve the readability of the code.